pkgs/development/python-modules/whenever/default.nix

```diff
@@ -14,38 +17,52 @@
 
 buildPythonPackage rec {
   pname = "whenever";
-  version = "0.5.2";
+  version = "0.6.1";
   pyproject = true;
 
-  disabled = pythonOlder "3.8";
+  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "ariebovenberg";
     repo = "whenever";
     rev = "refs/tags/${version}";
-    hash = "sha256-bG8LV+r5MjA1JwBHWy9/Io4daldAlyEGYNLW+5ITuOw=";
+    hash = "sha256-uUjQtaqPO/Ie7vVddQhc3dxORX2PxNRaDJzCr+vieUo=";
   };
 
-  postPatch = ''
-    # unrecognized arguments since we don't use pytest-benchmark in nixpkgs
-    substituteInPlace pytest.ini \
-      --replace-fail '--benchmark-disable' '#--benchmark-disable'
-  '';
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-8U3pGKY9UQ0JpzUn3Ny6YSD3wzXPDi1pupD5fpEJFvw=";
+  };
 
-  build-system = [ poetry-core ];
+  build-system = [
+    setuptools
+    setuptools-rust
+    rustPlatform.cargoSetupHook
+    cargo
+    rustc
+  ];
 
-  dependencies = [ tzdata ] ++ lib.optionals (pythonOlder "3.9") [ backports-zoneinfo ];
+  dependencies = [ tzdata ];
```

Comment:
I don't think we need `tzdata` for unix/linux.
https://github.com/ariebovenberg/whenever/blob/66592b4d218393772ce1bdfbd987ab231a8ece49/pyproject.toml#L29-L31